File: num2.rb
puts "full bottles: "
n = gets.chomp.to_i
puts "empty bottles for exchange: "
m = gets.chomp.to_i
mc = n
mcount = n
count = 0
ost = 0
drinkedBottles = 0

while mcount > 0
  puts "F" * n
  drinkedBottles += n
  count += 1
  mc = n + ost
  puts "E" * mc
  ost = mc % m
  n = mc / m
  count += 1
  mc = ost
  mcount = n
end

puts "#{drinkedBottles} #{count-1}"

File: num2.js
function main() {
  let n = 0;
  console.log("full bottles: ");
  n = parseInt(prompt());
  let m = 0;
  console.log("empty bottles for exchange: ");
  m = parseInt(prompt());
  let mc = n;
  let mcount = n;
  let count = 0;
  let ost = 0;
  let drinkedBottles = 0;
  while (mcount > 0) {
    console.log("F".repeat(n));
    drinkedBottles += n;
    count++;
    mc = n + ost;
    console.log("E".repeat(mc));
    ost = mc % m;
    n = Math.floor(mc / m);
    count++;
    mc = ost;
    mcount = n;
  }
  console.log(drinkedBottles + " " + (count - 1));
}

main();
